### PR TITLE
Feat/#286 projects filter

### LINKED ISSUE #286: Filter
Implements search and category filter
reworks filter of proposals page

### LINKED ISSUE #287: Search Project

### LINKED ISSUE #288: Category Filter

### LINKED ISSUE #289: Filter via URL

### LINKED ISSUE #291: Add Filter Response

### LINKED ISSUE #295: Unwanted initial loading of all proposals
When a user opens the proposal page with preselected filter options, we show all proposals for a short time before displaying the filtered list. This could be irritating for the user and we should prevent this behavior.

**AC:**  
- Prevent the "flickering" of the initial list
- Immediately show the filtered list

### PR DESCRIPTION
**Pull request by Zuckerwattederivat**
Resolves #286, resolves #295, resolves #291, resolves #289, resolves #287, resolves #288

### CODE REVIEW

File: components/app/projects/ProjectsDropdowns.vue [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    setItems(payload) {
+      this.watching = true;
+
+      const { id } = payload;
+
+      this.categoryItems.forEach((categoryItem) => {
+        // eslint-disable-next-line no-param-reassign
+        categoryItem.selected = false;
+        // eslint-disable-next-line no-param-reassign
+        if (categoryItem.id === id) categoryItem.selected = true;
```
Comment (KinimodEtlon) [inline components/app/projects/ProjectsDropdowns.vue:76]: - [ ] `categroyItem.selected = categoryItem.id === id` does the same in one line

File: components/app/projects/ProjectsFilter.vue [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      };
+
+      // replace history state
+      replaceQueryParams(this, this.filter);
+    },
+  },
+};
+</script>
+
+<style scoped></style>
```
Comment (KinimodEtlon) [inline components/app/projects/ProjectsFilter.vue:84]: - [ ] delete unnecessary code, do this in all other files that you worked on as well. E.g components/app/projects/ProjectsDropdowns.vue

File: components/app/projects/ProjectsList.vue [hunk 3]
```diff
@@ -1,69 +1,65 @@
...
+    "
+  >
+    <div v-for="project in projects" :key="project.id">
+      <NuxtLink
+        :prefetch="false"
+        :to="`/projects/${beautifyProjectId(project)}`"
+      >
+        <div class="shadow rounded p-2 pb-8 text-center">
+          <!--
             <div class="absolute top-0 right-0 mr-3 flex space-x-2">
```
Comment (KinimodEtlon) [inline components/app/projects/ProjectsList.vue:21]: - [ ] remove dead code

**Comment by KinimodEtlon** [comment]
Please do only one PR per Ticket, if you link 7 tickets, it's hard to get the whole scope of the PR. If you want to do more in one ticket, make the scope of the ticket bigger.

**Review by marhali — APPROVED** [review]

**Pull request merged**